pilot/cmd/pilot-agent/config.go

```diff
@@ -56,7 +58,17 @@ func constructProxyConfig() (meshconfig.ProxyConfig, error) {
 		proxyConfig = *meshConfig.DefaultConfig
 	}
 
-	proxyConfig.Concurrency = &types.Int32Value{Value: int32(concurrency)}
+	// If concurrency is unset, we will automatically set this based on CPU requests/limits for sidecars.
+	// For gateways, this will use all available CPUs.
+	// If explicitly set concurrency flag, this will be used.
+	if concurrency == 0 && role.Type == model.SidecarProxy {
+		byResources := determineConcurrencyOption()
+		if byResources != nil {
+			proxyConfig.Concurrency = byResources
+		}
```

Comment:
Will `proxyConfig.Concurrency` default to 0 if unset?

Reply:
The default is 2 which I think we want to keep as the default if nothing is explicitly set (no cpu resource files or flag set). If we set it to 0 here, it would be changing the behavior of someone just running `pilot-agent` standalone (I do this all the time for dev, but also occurs with VMs)